src/ReaderScreen/components/AudioPlayer/hooks/useAudioSyncScroll.js

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useRef, useState } from "react";
+import { useSelector } from "react-redux";
+
+const useAudioSyncScroll = (progress, isPlaying, webViewRef, audioUrl) => {
+  const isAudioSyncScroll = useSelector((state) => state.isAudioSyncScroll);
+  const lastSequenceRef = useRef(null);
+  const isScrollingRef = useRef(false);
+  const scrollTimeoutRef = useRef(null);
+  const [baniLRC, setBaniLRC] = useState(null);
+
+  // Function to fetch JSON file content
+  const fetchLRCData = async (jsonUrl) => {
+    try {
+      const response = await fetch(jsonUrl);
+
+      if (response.ok) {
+        const data = await response.json();
+        return data;
+      }
+      return false;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  // Load LRC data when audioUrl changes
+  useEffect(() => {
+    if (!audioUrl) {
+      setBaniLRC(null);
+      return;
+    }
+
+    const jsonUrl = audioUrl.replace(".mp3", ".json");
+    fetchLRCData(jsonUrl).then((data) => {
+      setBaniLRC(data);
```

Comment:
If component unmounts while fetch is in progress, it will try to set state on unmounted component.